proposals/config_ownership_crd.md

```diff
@@ -0,0 +1,129 @@
+  - Published: June 2nd, 2019
+  - Status:  Open for comments
+
+### Problem Statement
+As of this writing Ingress Controller (v0.6.0) blindly overwrites existing Application Gateway configurations when applying changes to:
+  - [listeners](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L100)
+  - [routing rules](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L107)
+  - [HTTP settings](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L83)
+  - [backend pools](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L90)
+  - [health probes](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L76)
+
+Ingress Controller assumes full ownership and control of the given App Gateway. This makes it impossible to deploy it
+to environments with existing pre-configured App Gateways (brownfield deployments). It is also not possible to use AGIC
+in hybrid environments where Application Gateway configuration is managed in parts by system administrators as well as
+Ingress Controller.
+
+### Proposed Solution
+We propose the creation of two new Kubernetes custom resource definitions ([CRD](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)):
+  - `AzureIngressManagedListener` - defines a reference to a single listener. Ingress Controller will assume ownership
+  and will mutate configuration for the listener and all related underlying resources: health probes, HTTP settings,
+  backend pools etc.
+  - `AzureIngressProhibitedListener` - defines a reference to a listener, which Ingress Controller is not permitted
+    to mutate. This listener and all related resources and configuration are assumed to be under control by another
+    system; Ingress Controller will not make any modifications to these.
+
+### Proposed CRD schema:
+Both `AzureIngressManagedListener` and `AzureIngressProhibitedListener` would have the same shape:
+
+##### AzureIngressManagedListener
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: azureingressmanagedlisteners.appgw.ingress.k8s.io
+spec:
+  group: appgw.ingress.k8s.io
+  version: v1
+  names:
+    kind: AzureIngressManagedListener
+    plural: azureingressmanagedlisteners
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            ip:
+              description: "(required) IP address of the listener managed by Ingress Controller; Could be the public or private address attached to the Application Gateway"
+              type: string
```

Comment:
`IPv4`/`IPv6` validation

Reply:
I agree it'd be great to have the validators here. I'd like to keep them out of the proposal for now.
Once we iron out the details of these CRDs - when we are ready to actually commit the YAML as a separate file - we will add validations.
I'm concerned that adding a long regex will distract from the overall goal of determining whether the CRD and its shape here make sense.